internal/service/pushbasedlogexport/state_transition_test.go

```diff
@@ -0,0 +1,166 @@
+package pushbasedlogexport_test
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	"go.mongodb.org/atlas-sdk/v20231115010/admin"
+	"go.mongodb.org/atlas-sdk/v20231115010/mockadmin"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/retrystrategy"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/pushbasedlogexport"
+)
+
+var (
+	activeState           = "ACTIVE"
+	unconfiguredState     = "UNCONFIGURED"
+	initiatingState       = "INITIATING"
+	bucketVerifiedState   = "BUCKET_VERIFIED"
+	assumeRoleFailedState = "ASSUME_ROLE_FAILED"
+	unknown               = ""
+	sc500                 = conversion.IntPtr(500)
+	currentTime           = time.Now()
```

Comment:
i think it's ok here but sometimes using the current (real) time can lead to flaky tests

Reply:
let me know if you have any suggestions?